strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/JwtAuthenticationTest.java

```diff
@@ -0,0 +1,120 @@
+
+import javax.inject.Inject;
+
+import java.util.Collections;
+
+import static io.restassured.module.mockmvc.RestAssuredMockMvc.given;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * @author: adavid9
+ */
+
+@IntegrationTest
+public class JwtAuthenticationTest
+        extends RestAssuredBaseTest
+{
+
+    private static final String UNAUTHORIZED_MESSAGE = "Full authentication is required to access this resource";
+
+    @Inject
+    private SecurityTokenProvider securityTokenProvider;
+
+    @Override
+    @BeforeEach
+    public void init()
+            throws Exception
+    {
+        super.init();
+        setContextBaseUrl(getContextBaseUrl() + "/api");
+    }
+
+    @Test
+    @WithAnonymousUser
+    public void testJWTAuthShouldPassWithToken()
+            throws Exception
+    {
+
+        String url = getContextBaseUrl() + "/users";
+
+        String basicAuth = "Basic YWRtaW46cGFzc3dvcmQ=";
+
+        String body = given().header(HttpHeaders.AUTHORIZATION, basicAuth)
+                             .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
+                             .when()
+                             .get(getContextBaseUrl() + "/login")
+                             .then()
+                             .statusCode(HttpStatus.OK.value())
+                             .extract()
+                             .asString();
+
+        // this token will expire after 1 hour
+        String tokenValue = getTokenValue(body);
+
+        given().header(HttpHeaders.AUTHORIZATION, tokenValue)
+               .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
+               .when()
+               .get(url)
+               .then()
+               .statusCode(HttpStatus.OK.value())
+               .body(notNullValue());
+
+    }
+
+    @Test
+    @WithAnonymousUser
+    public void testJWTAuthShouldFailWithoutToken()
+    {
+
+        String url = getContextBaseUrl() + "/users";
+
+        given().header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
+               .when()
+               .get(url)
+               .then()
+               .statusCode(HttpStatus.UNAUTHORIZED.value())
+               .body("error", equalTo(UNAUTHORIZED_MESSAGE));
+    }
+
+    @Test
+    @WithAnonymousUser
+    public void testJWTExpirationToken()
+            throws Exception
+    {
+
+        String url = getContextBaseUrl() + "/users";
+
+        // generate token that will expire after 1 second
+        String expiredToken = securityTokenProvider.getToken("admin", Collections.emptyMap(), 1);
+
+        Thread.sleep(1500);
+
+        given().header(HttpHeaders.AUTHORIZATION, expiredToken)
+               .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
+               .when()
+               .get(url)
+               .then()
+               .statusCode(HttpStatus.UNAUTHORIZED.value())
+               .body("error", equalTo(UNAUTHORIZED_MESSAGE));
```

Comment:
In case of expired token it should return `expired` message like assumed [here](https://github.com/strongbox/strongbox/blob/master/strongbox-security/strongbox-authentication-providers/strongbox-default-authentication-provider/src/main/java/org/carlspring/strongbox/authentication/api/impl/xml/JwtAuthenticationProvider.java#L52). 

@adavid9 can you please also check it?